tests/transcoding/index.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { transliterateRecursively } from "../../src/transcoding";
+import { isEncodableI8, encodeI8, decodeI8, transliterateI8 } from "../../src/transcoding/din66003drv"
+import { ValidationError, ValidationResultType } from "../../src/validation/index";
+import { payload3 } from "../samples/billingPayloads";
+
+
+describe("transcoding to I8", () => {
+
+    it("string is encodable to I8", () => {
+        expect(isEncodableI8("außergewöhnlich, spécial, særlig \n")).toEqual(true);
+    });
+
+    it("string is not encodable to I8", () => {
+        expect(isEncodableI8("außergewöhnlich, spécial, zwłaszcza")).toEqual(false);
+    });
+
+    it("encode string to I8", () => {
+        expect(encodeI8("außergewöhnlich, spécial, særlig \n"))
+        .toEqual(new Uint8Array([
+            97, 117, 126, 101, 114, 103, 101,
+            119, 124, 104, 110, 108, 105, 99,
+            104, 44, 32, 115, 112, 233, 99,
+            105, 97, 108, 44, 32, 115, 230,
+            114, 108, 105, 103, 32, 10
+        ]));
+    });
+
+    it("encode string to I8 and decode back to UTF-8", () => {
+        expect(decodeI8(encodeI8("außergewöhnlich, spécial, særlig \r\n\t")))
+            .toEqual("außergewöhnlich, spécial, særlig \r\n\t");
+    });
+
+    it("transliterate string for I8", () => {
+        expect(transliterateI8("Schöne Grüße an René François Lacôte, Małgorzata Dąbrowski, Nærøy, Борис Николаевич Ельцин, Δημήτρης Φωτόπουλος, אברהם הלוי פרנקל, Trần Hưng Đạo, สงขลา, கன்னியாகுமரி, महासमुंद, 深圳!\r\n\t"))
+            .toEqual("Schöne Grüße an René François Lacôte, Malgorzata Dabrowski, Nærøy, Boris Nikolaevich El'tsin, Dimitris Fotopoylos, 'vrhm hlvy frnkl, Tran Hung Dao, sngkhla, knniyakumri, mhasmumd, ShenZhen!\r\n\t");
+    });
+
+    it("recursively transliterate billing data and invoices for I8", () => {
+        const { transliterated, warnings } = transliterateRecursively(payload3, transliterateI8);
```

Comment:
I'm a bit uneasy about this `transliterateRecursively` and that you can apparently hand it any object to transliterate.

Where is the guarantee that it will not transliterate identifiers?

Reply:
It is guaranteed that the method transliterates only strings. As additional security measure, the method returns a warning for each string that is changed by the transliteration, so that the actual changes can be tracked transparently before transmitting the email.

An identifier like an IK does not contain characters that need to be transliterated. If it does, it would be an error and transliteration is required and the warning makes the problem visible.